core/src/main/scala/org/http4s/blaze/pipeline/stages/TimeoutStageBase.scala

```diff
@@ -52,6 +61,11 @@ abstract class TimeoutStageBase[T](timeout: Duration, exec: TickWheelExecutor) e
   final protected def cancelTimeout(): Unit = setAndCancel(null)
 
   final protected def startTimeout(): Unit = resetTimeout()
-
-
 }
+
+private object TimeoutStageBase {
+  // Represents the situation where the pipeline has been closed.
+  val closedTag = new Cancellable {
```

Comment:
It is a reference equality which should be indistinguishable for a singleton defined by a val or a case object.

I didn't use `Cancellable.noopCancel` because the purpose is more refined: it serves as a tag, not just as a noop. Additionally, making it private in the case object keeps the meaning local so future contributors (and myself) won't need to think about what `noopCancel` does, who else uses it, could it leak into the method call by some other use, etc.  
Although now that you mention i I will use `noopCancel` to replace the `null` references.